megamek/src/megamek/common/verifier/TestMech.java

```diff
@@ -589,21 +589,19 @@ public boolean checkCriticalSlotsForEquipment(Entity entity,
                 } else if (m.getType() instanceof PPCWeapon) {
                     weapon = "PPCs";
                 }
-                if (entity.hasSystem(Mech.ACTUATOR_LOWER_ARM, loc)
-                        || entity.hasSystem(Mech.ACTUATOR_HAND, loc)) {
-                    buff.append("Omni mechs with arm mounted " + weapon
-                            + " cannot have lower armor or hand actuators!\n");
+                if (mech.hasSystem(Mech.ACTUATOR_LOWER_ARM, loc)
+                        || mech.hasSystem(Mech.ACTUATOR_HAND, loc)) {
+                    buff.append("Omni mechs with arm mounted ").append(weapon)
+                            .append(" cannot have lower armor or hand actuators!\n");
                     legal = false;
                 }
             }
         }
-        if ((countInternalHeatSinks > engine.integralHeatSinkCapacity(mech
-                .hasCompactHeatSinks()))
-                || ((countInternalHeatSinks < engine
-                        .integralHeatSinkCapacity(mech.hasCompactHeatSinks()))
-                        && (countInternalHeatSinks != ((Mech) entity)
-                                .heatSinks()) && !entity.isOmni())) {
-            heatSinks.addElement(Integer.valueOf(countInternalHeatSinks));
+        if ((countInternalHeatSinks > engine.integralHeatSinkCapacity(this.mech.hasCompactHeatSinks()))
+                || ((countInternalHeatSinks < engine.integralHeatSinkCapacity(this.mech.hasCompactHeatSinks()))
+                        && (countInternalHeatSinks != mech.heatSinks(false))
```

Comment:
This is the only logic change in this file. The boolean parameter to Mech#heatSinks tells it to ignore prototypes.